client/core/core.go

```diff
@@ -1646,9 +1670,7 @@ func (c *Core) ReconfigureWallet(appPW, newWalletPW []byte, assetID uint32, cfg
 	c.wallets[assetID] = wallet
 	c.walletMtx.Unlock()
 
-	if oldWallet.connected() {
-		go oldWallet.Disconnect()
-	}
+	go oldWallet.Disconnect()
```

Comment:
`(*ConnectionMaster).Disconnect` is now safe to call in any state.  See the change in dex/runner.go